src/protocols/secure_channel/CASESession.cpp

```diff
@@ -339,57 +300,54 @@ CHIP_ERROR CASESession::DeriveSecureSession(SecureSession & session, SecureSessi
 
 CHIP_ERROR CASESession::SendSigmaR1()
 {
-    uint16_t data_len =
-        static_cast<uint16_t>(kSigmaParamRandomNumberSize + sizeof(uint16_t) + sizeof(uint16_t) +
-                              mOpCredSet->GetCertCount() * kTrustedRootIdSize + kP256_PublicKey_Length + sizeof(uint64_t) * 4);
+    uint16_t data_len = static_cast<uint16_t>(kSigmaParamRandomNumberSize + sizeof(uint16_t) + kSHA256_Hash_Length +
+                                              kP256_PublicKey_Length + sizeof(uint64_t) * 4);
```

Comment:
It would be good to have an "estimate struct overhead" method that takes data sizes and "number of fields overall" and does the heuristic above, so that the heuristic of "sizeof(uint64_t)-ish per TLV field should do" is clearer.